src/main/java/org/terasology/structureTemplates/internal/systems/StructureSpawnServerSystem.java

```diff
@@ -126,8 +127,11 @@ public void onSpawnStructureWithNoAnimation(SpawnStructureEvent event,
 
         for (RegionToFill regionToFill : spawnBlockRegionsComponent.regionsToFill) {
             Block block = regionToFill.blockType;
-
-            Region3i region = regionToFill.region;
+            if(entity.hasComponent(IgnoreAirBlocksComponent.class)&&block.getURI().getBlockFamilyDefinitionUrn().equals(BlockManager.AIR_ID.getBlockFamilyDefinitionUrn()))
```

Comment:
Please extract the check for air into its own method for better readbility:

```suggestion
            if(entity.hasComponent(IgnoreAirBlocksComponent.class) && isAir(block)) {
```